src/Bem2d.jl

```diff
@@ -26,17 +26,81 @@ function obsgrid(xmin, ymin, xmax, ymax, npts)
     return xobs[:], yobs[:]
 end
 
+export Element
+struct Element
+    x1::Float64
+    y1::Float64
+    x2::Float64
+    y2::Float64
+    a::Float64
+    b::Float64
+    normalstress::Float64
+    # The following are derived properties (could be dynamic)
+    angle::Float64
+    length::Float64
+    halflength::Float64
+    xcenter::Float64
+    ycenter::Float64
+    rotmat::Array{Float64,2}
+    rotmatinv::Array{Float64,2}
+    xnormal::Float64
+    ynormal::Float64
+    xnodes::Array{Float64,1}
+    ynodes::Array{Float64,1}
+    # These are unused?
+    # uxconst::Float64
+    # uyconst::Float64
+    # uxquad::Array{Float64,1}
+    # uyquad::Array{Float64,1}
+end
+function Element(; x1, y1, x2, y2, a, b, normalstress)
+    dx = x2 - x1
+    dy = y2 - y1
+    magnitude = sqrt(dx^2 + dy^2)
+    angle = atan(dy, dx)
+    length = magnitude
+    halflength = 0.5 * length
```

Comment:
Some of these simpler "derived" properties could easily be dynamically computed on-demand instead of stored.  You'd then either use an accessor-like function or if you want to be fancy you could overload [`getproperty`](https://docs.julialang.org/en/v1/base/base/#Base.getproperty).

Reply:
```Julia
halflength(e::Element) = 0.5 * e.length
```

Instead of accessing `e.halflength`, you'd access `halflength(e)`.  One nice advantage is it's easier to use broadcasting: `halflength.(e)`.